lib/active_record/connection_adapters/percona_adapter.rb

```diff
@@ -5,12 +5,14 @@
 require 'forwardable'
 
 module ActiveRecord
-  class Base
+  module ConnectionHandling
     # Establishes a connection to the database that's used by all Active
     # Record objects.
-    def self.percona_connection(config)
+    def percona_connection(config)
       mysql2_connection = mysql2_connection(config)
 
+      config[:username] = 'root' if config[:username].nil?
```

Comment:
This is the MySQL standard; when no user is provided root is used by default. Copied exactly from the MySql2Adapter